receiver/oracledbreceiver/config.go

```diff
@@ -0,0 +1,54 @@
+// Copyright Splunk, Inc.
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oracledbreceiver // import "github.com/signalfx/splunk-otel-collector/receiver/oracledbreceiver"
+
+import (
+	"errors"
+	"net/url"
+	"time"
+
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/signalfx/splunk-otel-collector/receiver/oracledbreceiver/internal/metadata"
+)
+
+type Config struct {
+	config.ReceiverSettings                 `mapstructure:",squash"`
+	DataSource                              string `mapstructure:"datasource"`
```

Comment:
Not necessary in first pass but I can see also providing other options and using [`BuildUrl()`](https://github.com/sijms/go-ora/blob/master/v2/connection_string.go#L123) as convenient alternative for the user.